flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/expressions/DefaultExpressionVisitor.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.expressions;
+
+/**
+ * A utility {@link ApiExpressionVisitor} that calls {@link DefaultExpressionVisitor#defaultMethod(Expression)} by
+ * default, unless other methods are overriden explicitly.
+ */
+public abstract class DefaultExpressionVisitor<ResultT> extends ApiExpressionVisitor<ResultT> {
```

Comment:
nit: Can we make our default visitors consistent? See `org.apache.flink.table.plan.util.RexDefaultVisitor`. Call it `ApiExpressionDefaultVisitor` with a `visitExpression` method?